pkg/controller/podautoscaler/replica_calculator_test.go

```diff
@@ -317,34 +317,40 @@ func (tc *replicaCalcTestCase) runTest(t *testing.T) {
 		assert.Equal(t, tc.resource.expectedUtilization, outUtilization, "utilization should be as expected")
 		assert.Equal(t, tc.resource.expectedValue, outRawValue, "raw value should be as expected")
 		assert.True(t, tc.timestamp.Equal(outTimestamp), "timestamp should be as expected")
+		return
+	}
 
-	} else {
-		var outReplicas int32
-		var outUtilization int64
-		var outTimestamp time.Time
-		var err error
-		if tc.metric.singleObject != nil {
-			outReplicas, outUtilization, outTimestamp, err = replicaCalc.GetObjectMetricReplicas(tc.currentReplicas, tc.metric.targetUtilization, tc.metric.name, testNamespace, tc.metric.singleObject, selector)
-		} else if tc.metric.selector != nil {
-			if tc.metric.targetUtilization > 0 {
-				outReplicas, outUtilization, outTimestamp, err = replicaCalc.GetExternalMetricReplicas(tc.currentReplicas, tc.metric.targetUtilization, tc.metric.name, testNamespace, tc.metric.selector, selector)
-			} else if tc.metric.perPodTargetUtilization > 0 {
-				outReplicas, outUtilization, outTimestamp, err = replicaCalc.GetExternalPerPodMetricReplicas(tc.currentReplicas, tc.metric.perPodTargetUtilization, tc.metric.name, testNamespace, tc.metric.selector)
-			}
-		} else {
-			outReplicas, outUtilization, outTimestamp, err = replicaCalc.GetMetricReplicas(tc.currentReplicas, tc.metric.targetUtilization, tc.metric.name, testNamespace, selector)
-		}
+	var outReplicas int32
+	var outUtilization int64
+	var outTimestamp time.Time
+	matchedCases := 0
+	if tc.metric.singleObject != nil {
```

Comment:
make this a switch statement

Reply:
When writing a test you can specify both tc.metric.targetUtilization and  tc.metric.perPodTargetUtilization.

The old version of the code woukd just ignore one of them. I think it's a mistake and so the test will fail.

Should I improve description of the commit / add some comments?

Reply:
You're not supposed to be able to specify both, IIRC.  You're supposed to be basically chosing which of the replica calc functions you want to test depending on the combination of test parameters that you set.  You don't want a single test testing multiple different replica calc functions, since it makes it harder to see what the actual failure was.

That being said, we should probably make it a bit clearer what test parameters choose which functions.

Reply:
Done, I think.